### ISSUE TITLE
DRBD 9.1.12: Quorum lost on secondary

### ISSUE THREAD

**Issue by phoenix-bjoern** [LINBIT/drbd#52]
Several resources are shown with status "quorum lost" in drbdmon but only on secondaries and only on one node. IO is working fine.

This is the output of `drbdsetup show`:

```
root@de-fra-node11:/# drbdsetup show pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5
resource "pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5" {
    options {
        on-no-data-accessible   suspend-io;
        quorum                  majority;
        quorum-minimum-redundancy       2;
        on-suspended-primary-outdated   force-secondary;
    }
    _this_host {
        node-id                 1;
        volume 0 {
            device                      minor 1176;
            disk                        "/dev/vg-pool/pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5_00000";
            meta-disk                   internal;
            disk {
                rs-discard-granularity  65536; # bytes
            }
        }
    }
    connection {
        _peer_node_id 2;
        path {
            _this_host ipv4 192.168.10.214:7176;
            _remote_host ipv4 192.168.11.191:7176;
        }
        net {
            cram-hmac-alg       "sha1";
            shared-secret       "JLYRWkUgIJjidWuQ2DTX";
            rr-conflict         retry-connect;
            verify-alg          "sha1";
            max-buffers         12288;
            _name               "de-fra-node54";
        }
    }
    connection {
        _peer_node_id 0;
        path {
            _this_host ipv4 192.168.10.214:7176;
            _remote_host ipv4 192.168.11.193:7176;
        }
        net {
            cram-hmac-alg       "sha1";
            shared-secret       "JLYRWkUgIJjidWuQ2DTX";
            rr-conflict         retry-connect;
            verify-alg          "sha1";
            max-buffers         12288;
            _name               "de-fra-node56";
        }
    }
}
```

And this is the output of `drbdadm status`:

```
root@de-fra-node11:/# drbdadm status pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5
pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5 role:Secondary suspended:quorum
  disk:UpToDate quorum:no blocked:upper
  de-fra-node54 role:Primary
    peer-disk:UpToDate
  de-fra-node56 role:Secondary
    peer-disk:UpToDate
```

This is `drbdmon`:

```
⤷RES: pvc-96c2dc8b-31aa-4b28-8ab7-332b58de64b5          Secondary   QUORUM LOST
    ✗    0:   1176  UpToDate            
    ⚬⇄⚫de-fra-node54 ⚬⇄ de-fra-node56
```

On the other peers drbdadm reports no issues.

Why does node11 show a quorum issue while node54+56 show the disk as "UpToDate"? Shouldn't the peers show the issue as well? And shouldn't DRBD try to recover from the "no quorum" state automatically?

N.B. In Linstor the volume is shown as "UpToDate" on all nodes while drbdmon is reporting "QUORUM LOST" which makes detecting pretty difficult.

**Comment by bmalynovytch** [comment]
Adding more diskless peers makes quorum happy again 🤔

**Comment by phoenix-bjoern** [comment]
@bmalynovytch no, that doesn’t solve the problem. As noted: the primary had quorum, only one of the secondaries showed „no quorum“. That’s why I think it is simply a bug. 
But will keep and on eye on this. Haven’t seen the issue for a while, maybe the problem in DRBD has been resolved in one of the latest versions.

**Comment by bmalynovytch** [comment]
You're right @phoenix-bjoern , it's only hitting secondaries (diskful), primary (diskless) is ok.

**Comment by WanzenBug** [comment]
I suspect this is related to issues introduced with 3c789500c8e7b3da46e6477a3d351de99652c78d

We saw some cases where this migration did not trigger as one would expect. I believe this is to be fixed in 9.2.2/9.1.13

**Comment by phoenix-bjoern** [comment]
Thanks @WanzenBug for the insight. Will close the ticket as I haven't seen this issue for a while. Will open it again if it still occurs with 9.1.13.
